### PR TITLE
V3/migrate icons

### PR DESCRIPTION
**Pull request by ugogo**
# Migrate `Icon` component

## Breaking changes
- Icon sizes and list are not available anymore via the component (`Icon.ICONS.x`, `Icon.SIZES.x`)

## Improvements
- The `size` prop now accept a custom value, in addition to the current `s|m|l` ones

## Notes
- The previous `Icon` component has been renamed `_DeprecatedIcon` to not break other components implementation

### CODE REVIEW

File: .storybook/theme/icons.js [hunk 1]
```diff
@@ -0,0 +1,1309 @@
+import * as React from 'react';
```
Comment (ugogo) [inline .storybook/theme/icons.js:1]: This will be moved later on with the `ThemeProvider` component

File: scripts/icons.js [hunk 2]
```diff
@@ -88,10 +82,16 @@ const formatIcons = iconsPath.map(async iconPath => {
...
+  const output = `import * as React from 'react';
+
+/*
+ * This file has been automatically generated.
+ * Do not update it manually, run 'yarn icons' instead.
+ */
 
-/* This file has been generated via the 'icons' script */
-${reactIcons}`;
+export default {
```
Comment (ugogo) [inline scripts/icons.js:92]: Icons are not in an object, rather than in a React component

File: src/components/Icon/Icon.stories.tsx [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+import Icon, { IconSize } from './Icon';
+import Input from '../Input';
+import Label from '../Label';
+
+type IconsFiltererProps = {
+  children: (icons: string[]) => React.ReactNode;
+  filter?: string;
+};
+
+const IconsLister = ({ children, filter = '' }: IconsFiltererProps) => {
```
Comment (ugogo) [inline src/components/Icon/Icon.stories.tsx:18]: For some reasons, the `useTheme` hook wasn't returning the theme inside a story 🤷

Comment (obedparla) [reply src/components/Icon/Icon.stories.tsx:18]: is the story wrapped in a ThemeProvider?

**Comment by ugogo** [comment]
> Why are all the icons imported from _DeprecatedIcon instead of the new Icon, is there some limitation for now to do it that way, or it's just to avoid changing a bunch of other components?

Because I didn't want to introduce too many changes
The DeprecatedIcon component will be removed when refactoring the wrapping components

File: src/components/Icon/Icon.stories.tsx (hunk 3 above)
Comment (ugogo) [reply src/components/Icon/Icon.stories.tsx:18]: Yes, cf: https://github.com/heetch/flamingo/pull/158/files#diff-41e445b678933963d77632d4206c90b8R17
It doesn't work only when the hook is used at the top level 🤷

**Pull request merged**